sdk/src/asset_handlers/font_io.rs

```diff
@@ -393,76 +402,129 @@ impl TableHead {
         if size != actual_size {
             Err(Error::FontLoadError)
         } else {
-            Ok(Self {
+            let head = Ok(Self {
+                // 0x00
                 majorVersion: reader.read_u16::<BigEndian>()?,
                 minorVersion: reader.read_u16::<BigEndian>()?,
+                // 0x04
                 fontRevision: reader.read_u32::<BigEndian>()?,
+                // 0x08
                 checksumAdjustment: reader.read_u32::<BigEndian>()?,
                 // TBD - FontLoadError when head magic is bad?
                 magicNumber: reader.read_u32::<BigEndian>()?,
+                // 0x10
                 flags: reader.read_u16::<BigEndian>()?,
                 unitsPerEm: reader.read_u16::<BigEndian>()?,
+                // 0x14
                 created: reader.read_i64::<BigEndian>()?,
+                // 0x1c
                 modified: reader.read_i64::<BigEndian>()?,
+                // 0x24
                 xMin: reader.read_i16::<BigEndian>()?,
                 yMin: reader.read_i16::<BigEndian>()?,
+                // 0x28
                 xMax: reader.read_i16::<BigEndian>()?,
                 yMax: reader.read_i16::<BigEndian>()?,
+                // 0x2c
                 macStyle: reader.read_u16::<BigEndian>()?,
                 lowestRecPPEM: reader.read_u16::<BigEndian>()?,
+                // 0x30
                 fontDirectionHint: reader.read_i16::<BigEndian>()?,
                 indexToLocFormat: reader.read_i16::<BigEndian>()?,
+                // 0x34
                 glyphDataFormat: reader.read_i16::<BigEndian>()?,
-            })
+                // 0x36 - 54 bytes
+                // TBD - Two bytes of padding to get to 56/0x38. Should we
+                // seek/discard two more bytes, just to leave the stream in a
+                // known state? Be nice if we didn't have to.
+                //   1. On the one hand, whoever's invoking us could more-
+                //      efficiently mess around with the offsets and padding.
+                //   B. On the other, for the .write() code, we definitely push
+                //      the "pad *yourself* up to four, impl!" approach
+                //   III. Likewise the .checksum() code (although, because this
```

Comment:
Also, via [OTF spec](https://learn.microsoft.com/en-us/typography/opentype/spec/otff#calculating-checksums), I think you _do_ necessarily always include the pad bytes in the checksum (though, like you mentioned, as 0, they shouldn't matter) - _Table checksums are the unsigned sum of the uint32 units of a given table_

Reply:
(even more - it suggests you *assume* pad bytes are 0, even if they aren't - _To accommodate data with a length that is not a multiple of four, the above algorithm must be modified to treat the data as though it contains zero padding to a length that is a multiple of four._)

Reply:
As you say. At least it simplifies matters (checksum functions don't have to pretend to see zeros that aren't there).